Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
+release year, and review score.
+For simplicitly, just use column names in the table declaration:
+thanks to the `flexible typing`_ feature of SQLite,
+specifying the data types is optional.
+Execute the ``CREATE TABLE`` statement
+by calling :meth:`con.execute() <Cursor.execute>`::
+
+   cur.execute("CREATE TABLE movie(title, year, score)")
+
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but earlier versions of SQLite do not recognise that variant.
+
+You can verify that the new table has been created by querying
+the ``sqlite_master`` table built-in to SQLite.
+It should contain an entry for the ``movie`` table definition.
+Execute that query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+store the result in a variable ``res``,
+and call :meth:`res.fetchone() <Cursor.fetchone>` to fetch the first
+(and only) row that was returned::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master")
+   >>> res.fetchone()
+   ('movie',)
+
+As expected, the query shows the table is now created
+by returning a single :class:`tuple`: with the name of the table.
+As an exercise, try querying ``sqlite_master``
+for a non-existent table ``"abc"``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master WHERE name='abc'")
+   >>> res.fetchone()
+   >>>
+
+As expected, the query returns an empty result.
+
+Now, you'll add two rows of data supplied as SQL literals
+by executing an ``INSERT`` statement,
+once again by calling :meth:`cur.execute(...) <Cursor.execute>`::
+
+   cur.execute("""
+       INSERT INTO movie VALUES
+           ('Monty Python and the Holy Grail', 1975, 8.2),
+           ('And Now for Something Completely Different', 1971, 7.5)
+   """)
+
+The ``INSERT`` statement implicitly opens a transaction,
+which needs to be committed before changes are saved in the database
+(see :ref:`sqlite3-controlling-transactions` for details).
+Call :meth:`con.commit() <Connection.commit>` on the connection object
+to commit the transaction::
 
-   # Save (commit) the changes
    con.commit()
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+You can verify that the data was inserted correctly
+by executing a ``SELECT`` query.
+Use the now-familiar :meth:`con.execute(...) <Cursor.execute>` to
+store the result in ``res``,
+and call :meth:`res.fetchall() <Cursor.fetchall>` to fetch all rows::
 
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
-
-   import sqlite3
-   con = sqlite3.connect('example.db')
-   cur = con.cursor()
+   >>> res = cur.execute("SELECT score FROM movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
 
-At this point, our database only contains one row::
+The result is a :class:`list` of two :class:`!tuple`\s, one per row,
+each containing a the ``score`` from the query.
```

Comment:
```suggestion
The result is a :class:`list` of two :class:`!tuple`\s, one per row,
each containing the ``score`` from the query.
```
I like this phrasing.  What I wanted to convey is that tuples correspond to rows, and the tuple items to (the selected) columns).  Even though it's not explicitly stated, I think it's clear enough.

Reply:
To that end, it should be "a" [single] score, not "the" score from the query which would imply there is only a single score per query, thus multiple queries (one per row) rather than multiple scores

```suggestion
The result is a :class:`list` of two :class:`!tuple`\s, one per row,
each containing a ``score`` from the query.
```

Reply:
Here "score" is referring to the `score` column selected in the query (hence the ` ``...`` `), not the numeric value(s) found in the tuple(s).

Reply:
Yes, but again, using "the" in "each containing _the_ score from the query", implies there is one query per row, which is not the case. 

I drew that initial conclusion from a casual read of the sentence, but breaking it down from a semantic perspective, we can see why a reader could be confused. In this context, "the `score`" can't be referring to the whole column (the thing there is only one of in the query) as opposed to the _value_ of the column for each row (of which there is one per row, but multiple per query), because "_each_" tuple (thus each row, as stated) is stated to contain exactly one of "the score". Then, while there _is_ "_the_ score from the _row_", what is is stated is "_the_ score from the _query_", which means there must exactly one of these _per-row_ scores per one query. Therefore, this directly implies there is one query _per row_, which is of course not the case.

However, you _could_ also reword this, perhaps more clearly:

```suggestion
The result is a :class:`list` of two :class:`!tuple`\s, one per row,
each containing the ``score`` from the row.
```

Reply:
How about this (diff from a few lines above):

```diff
-and call :meth:`res.fetchall() <Cursor.fetchall>` to fetch all rows::
+and call :meth:`res.fetchall() <Cursor.fetchall>` to
+return a :class:`list` of all resulting rows::
 
    >>> res = cur.execute("SELECT score FROM movie")
    >>> res.fetchall()
    [(8.2,), (7.5,)]
 
+As expected, the resulting :class:`!list` contains two rows
+as :class:`!tuple`\s of ``(score,)``.
```

Reply:
IMO, making the structure explicit could be helpful, but I'm not sure splitting up and/or duplicating the discussion of the return type before and after the example makes sense. What about this?

```suggestion
The result is a :class:`list` of two :class:`!tuple`\s, one per row,
each containing that row's ``score`` value.
```

"As expected" seems to put an expectation on the student that they will know _a priori_ what will be returned, which I'm not sure is very Diataxis-y given what @evildmp has said

Reply:
SGTM